### ISSUE TITLE
Electron Support

### ISSUE THREAD

**Issue by adumbidiot** [napi-rs/napi-rs#125]
While experimenting with this library, I realized that I can't get any generated modules to be loaded by Electron on Windows. I'm running on Windows 10 and trying to use Electron 9.1.1. 

I think the problem is with linking to the wrong version of node.lib, as the build script appears to unconditionally link to the node.lib of the installed node, while it should be linking to electron's node.lib. I think this library is also missing a delay-load hook implementation.

**Comment by wspl** [comment]
Use this library with electron on macOS works well.

**Comment by adumbidiot** [comment]
Windows is a much more finicky platform and will require more work to support. There are a few problems with this library that I've found.

1. It is missing a delay-load implementation.
2. It doesn't respect npm environment variables in the build script.
3. It uses a primitive js script to download node.lib, which breaks under simple circumstances such as trying to follow redirects for electron's node.lib.
4. The global cache for node.libs is sperated by version. However, Electron versions their node.lib by electron version, not nodejs version. This will lead to builds failing unpredictably if someone builds for Electron and node on the same computer.
5. The linker args emitted by the build script appear to not be respected. I was able to find a solution only by moving the build script execution into the actual module's build script.
6. The napi version feature is set by the executing nodejs, not the engine its targeting.

I will open a PR to try to address these concerns to add proper Electron support. I can only test on Windows however, but maybe I can add a test suite for Electron on CI.
